src/karmen_backend/openapi/swagger.yaml

```diff
@@ -0,0 +1,1544 @@
+
+
+  '/organizations/{org_uuid}/gcodes':
+    get:
+      summary: 'List gcodes'
+      operationId: server.routes.gcodes.gcodes_list
+      tags:
+        - gcodes
+      parameters:
+        - $ref: '#/components/parameters/organization_uuid_param'
+      responses:
+        '200':
+          description: 'List of gcodes'
+    post:
+      summary: 'Upload gcode'
+      operationId: server.routes.gcodes.gcode_create
+      tags:
+        - gcodes
+      parameters:
+        - $ref: '#/components/parameters/organization_uuid_param'
+      requestBody:
+        content:
+          text/plain:
+            schema:
+              type: string
+              format: binary
+      responses:
+        '200':
+          description: 'Created'
+  '/organizations/{org_uuid}/gcodes/{gcode_uuid}':
+    get:
+      summary: 'Get gcode detail'
+      operationId: server.routes.gcodes.gcode_detail
+      tags:
+        - gcodes
+      parameters:
+        - $ref: '#/components/parameters/organization_uuid_param'
+        - $ref: '#/components/parameters/gcode_uuid_param'
+      responses:
+        '200':
+          description: 'Gcode details'
+    delete:
+      summary: 'Delete gcode'
+      operationId: server.routes.gcodes.gcode_delete
+      tags:
+        - gcodes
+      parameters:
+        - $ref: '#/components/parameters/organization_uuid_param'
+        - $ref: '#/components/parameters/gcode_uuid_param'
+      responses:
+        '204':
+          description: 'Deleted'
+  '/organizations/{org_uuid}/gcodes/{gcode_uuid}/data':
+    get:
+      summary: 'Download gcode file'
+      operationId: server.routes.gcodes.gcode_file
+      tags:
+        - gcodes
+      parameters:
+        - $ref: '#/components/parameters/organization_uuid_param'
+        - $ref: '#/components/parameters/gcode_uuid_param'
+      responses:
+        '200':
+          description: 'Gcode file'
+  '/octoprint-emulator/api/version':
+    get:
+      summary: 'Get octoprint version'
+      operationId: server.routes.octoprintemulator.version
+      tags:
+        - octoprint-emulator
+      responses:
+        '200':
+          description: ''
+  '/octoprint-emulator/api/settings':
+    get:
+      summary: 'Get octo settings'
+      operationId: server.routes.octoprintemulator.settings
+      tags:
+        - octoprint-emulator
+      responses:
+        '200':
+          description: ''
+  '/octoprint-emulator/api/printer':
+    get:
+      summary: 'Get printer status'
+      operationId: server.routes.octoprintemulator.printer
+      tags:
+        - octoprint-emulator
+      responses:
+        '200':
+          description: ''
+  '/octoprint-emulator/api/job':
+    get:
+      summary: 'Get job status'
+      operationId: server.routes.octoprintemulator.job
+      tags:
+        - octoprint-emulator
+      responses:
+        '200':
+          description: ''
+  '/octoprint-emulator/api/files/local':
+    post:
+      summary: 'Upload gcode'
```

Comment:
Upload gcode to a printer? But not sure actually what this does

Reply:
For uploading gcodes from slicers with `send gcode to octoprint` functionality. I described the emulator endpoints better.